### DISCUSSION TITLE
Pull Requests to Github Classroom assignments are visible to all students

### DISCUSSION THREAD

**Discussion by tedMcleod** [community/community#112506] [GitHub Education]
### Hi everyone,

I opted-in to the new forking system for Github Classroom assignments, but I have discovered some big downsides:

1. After students accept the assignment and they go to github desktop to clone their project, the template repository is listed as one of the repositories they can clone. This can lead to confusion.
2. Students who don't know what they are doing, will be prompted by online tutorials or other help pages to make a pull request to the template assignment. Within one day, a student made a pull request containing their complete solution to the problem. This is a HUGE problem because all students can go to the template assignment and see the pull request. The student closed their request, but it is still visible because there is no way to delete it.

**Comment by adarsh-site** [comment]
Here are some steps you can take to address these issues:

Clarify Instructions: Ensure that your instructions explicitly state that students should not interact with the template repository in any way, including cloning it or making pull requests to it. Emphasize the importance of working only with their forked repository.

Provide Guidance: Offer specific guidance on how students should clone their repositories using GitHub Desktop. You can create a step-by-step guide or video tutorial to walk them through the process and highlight the importance of selecting their forked repository, not the template.

Educate on Pull Requests: Educate students on the purpose and proper use of pull requests. Explain that pull requests should only be made when they want to propose changes to a repository, and they should always double-check that they're targeting the correct repository.

Address Pull Request Issue: Since a student has already made a pull request to the template assignment, you'll need to address this issue promptly. You can close the pull request from your end and communicate to your students why it's important to avoid interacting with the template repository.

Provide Support: Be available to answer any questions or concerns students may have about the assignment setup or GitHub workflow. Encourage them to reach out if they encounter any issues or confusion.

Report Feedback to GitHub: If you believe there are significant usability issues or flaws in the forking system that need to be addressed, consider providing feedback to GitHub. They may be able to implement improvements or provide guidance on best practices for using the forking system in educational settings.

**Comment by RyanHecht** [comment]
Hi @tedMcleod,

Thanks for your feedback! We're working on mitigation plans for this before we go to general availability

**Reply by ssardina** [reply]
@RyanHecht any update on this? I am a bit scared as June 17 is approaching and I am starting my courses soon.. 

There is another discussion rising the same concerns: #125414

**Comment by ssardina** [comment]
Hi @tedMcleod @RyanHecht . 

I have also opted-in and tried on assessment as a fork. I have been using the template methods for years and pushing any updates myself manually via a script; indeed not great but ...

Besides the two issues  described by @tedMcleod , I add the following one:

3. because students have read access automatically to the start code repo now, the teaching staff are not able to use that repo issue tracker privately anymore, as all students can see the tracker. We used the issue tracker not just to record glitches of the assessment, but also to track particular student issues or markings.
4. while it is very nice that students can sync their repos to the updated template one; we need to trust they will do it. It would be nice if Google Classroom provides a mechanism for teachers to be able to SYNC all forks in one shot. This may be tricky feature I suppose.

Of course I can always go to the template repo (the one containing the start code) and remove every student from having READ access, but this of course precludes them to SYNC their copies.

At this point, items  2+3  above are enough reasons for me to much prefer the (old) template approach. Can we at least do not remove the option of that approach until we find robust ways around these issues?

Thanks & keep up the good job; I love all the GH Classroom Infrastructure! :clap: :clap: :clap:

**Reply by ssardina** [reply]
BTW message in GH Classroom seems to suggest that the template approach will be removed in June 2024:

![image](https://github.com/community/community/assets/2368384/d0f453c6-ab45-4235-800c-ce30968fcd6e)

**Reply by ssardina** [reply]
Similar concerns were rose in #125414, any update on these issues? :pray:

**Comment by ssardina** [comment]
FYI for whoever is reading this thread, there seems to have been a resolution for all our concerns: #127455
